### DISCUSSION TITLE
Defaulting to /bin/sh.

### DISCUSSION THREAD

**Discussion by ghost** [coder/code-server#4475] [Q&A]
When opening a new terminal it defaults to `/bin/sh` I have set the default shell everwhere else to `/bin/bash` but code-server keeps using sh. Is there a way to change this?

OS: Ubuntu 20.04 LTS

**Comment by jsjoeio** [comment]
I'm actually running into this exact same issue in Coder (our product) 🤔 (which is on 3.10.2).

What happens if you run `echo $SHELL` in the Integrated Terminal?

cc @code-asher

**Reply by code-asher** [reply]
Yeah I am not completely sure how VS Code decides the terimanl to spawn but I believe it will read the `SHELL` env var.  There is also a setting within VS Code that can be configured.

**Reply by jsjoeio** [reply]
Hmm..maybe you need to overwrite the `SHELL` env var as well (because I have the VS Code settings configured properly yet still run into this). I'll have to look

**Reply by lg-nineoneninethreenine** [reply]
I think, there might be a `~/.shrc` file or any SH config file in the system. If there is a file like that, at the end of the file(`echo bash >> ~/.shrc`, add `bash` and source the file. I expect a shell file config or if there is not, maybe try:
1. Deleting the code-server data (might work)
2. Make the default shell as `bash` globally (using `chsh -s /bin/bash`)
3. Find a `sh` config and add `bash` to it.(most unlikely)

**Reply by jsjoeio** [reply]
Next time I run into this, I'll have to give it a try! Thanks for the suggestion @im-coder-lg !
